### ISSUE TITLE
Wrong OLS redirects from cellosaurus

### ISSUE THREAD

**Issue by matentzn** [biopragmatics/bioregistry#1259]
Redirects like this ones exist in bioregistry:

From `CL` to 

```
https://www.ebi.ac.uk/ols4/ontologies/clo/classes?iri=http://purl.obolibrary.org/obo/
```

This makes no sense because resolving this would lead to a bogus redirect:

`CL:123` ---> `https://www.ebi.ac.uk/ols4/ontologies/clo/classes?iri=http://purl.obolibrary.org/obo/123`

The correct URI redirect should be:

```
https://www.ebi.ac.uk/ols4/ontologies/clo/classes?iri=http://purl.obolibrary.org/obo/CL_
```

Note the CL in the end.

For context see https://github.com/mapping-commons/sssom-py/pull/557/files#r1827656744

@cthoyt thinks they may be coming in from the cellosaurus ingest.

@ddtxra @AmosBairoch can you help us determine who's the responsible person to contact at the moment?

**Comment by cthoyt** [comment]
Please take a look at https://github.com/biopragmatics/bioregistry/blob/481f50a6c988cbad112968936dde461c293aea67/src/bioregistry/external/cellosaurus/processed.json for context. There are three issues where the mixed use of an OBO IRI space is causing this issue:

https://github.com/biopragmatics/bioregistry/blob/481f50a6c988cbad112968936dde461c293aea67/src/bioregistry/external/cellosaurus/processed.json#L50-L54

https://github.com/biopragmatics/bioregistry/blob/481f50a6c988cbad112968936dde461c293aea67/src/bioregistry/external/cellosaurus/processed.json#L131-L136

https://github.com/biopragmatics/bioregistry/blob/481f50a6c988cbad112968936dde461c293aea67/src/bioregistry/external/cellosaurus/processed.json#L143-L148


What's going on is that Cellosaurus is conflating strings that look like `CLO_1234567` as CURIEs that get formatted into these strings. The correct way to write a URI format string is not to use the CURIE, but the local unique identifier, so instead, these should look like:

1. http://www.ontobee.org/ontology/BCGO?iri=http://purl.obolibrary.org/obo/BCGO_
2. http://www.ontobee.org/ontology/BCGO?iri=http://purl.obolibrary.org/obo/CL_
3. http://www.ontobee.org/ontology/BCGO?iri=http://purl.obolibrary.org/obo/CLO_

If there's a way I can make a drive-by curation, I would do so, but I think these are all handled internally at SIB these days. Please let me know if that's not the case and I will send a fix

**Comment by matentzn** [comment]
@cthoyt if we dont get a response, is there a way I can somehow exclude them from the obo context EPM?

**Comment by cthoyt** [comment]
@matentzn yes, the processing script is here https://github.com/biopragmatics/bioregistry/blob/main/src/bioregistry/external/cellosaurus/__init__.py. You can add a ruleset for fixing these

**Comment by matentzn** [comment]
@cthoyt despite "fixing" this issue, the bad URLS still show up:

```
"http://www.ebi.ac.uk/ols4/ontologies/bto/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252F",
```

in the EPM: 

https://github.com/mapping-commons/sssom-py/pull/568/files#diff-704e75fae3e22a017d381724e2ce3d8f9dabc38e4bb2cc39f05597a8bdfb93a8R403

**Comment by cthoyt** [comment]
The previous fix wasn't generic enough to catch the string you posted, since it has URL-encoded slashes. This conditional can be extended:

https://github.com/biopragmatics/bioregistry/blob/087bba56622bc82d057ca32b773891b987b18c43/src/bioregistry/external/cellosaurus/__init__.py#L79-L85

Related: https://github.com/biopragmatics/bioregistry/pull/1095 was a first step towards more detailed debugging on where different parts of the rasterized bioregistry data come from, in case errors like this are originating from hard to find external registry values
